src/Client.php

```diff
@@ -76,4 +82,112 @@ public function iterate(string $method, array $arguments): iterable
+     *     ['path' => '/tmp/image.png', 'title' => 'Bot Logo', 'alt_text' => 'Slack Bot Logo'],
+     *     ['path' => '/tmp/script.js', 'snippet_type' => 'javascript']
+     * ], 'C12345678', 'Here are the uploaded files:');
+     */
+    public function filesUploadV2(
+        array $files,
+        string $channelId,
+        ?string $initialComment = null,
+    ): FilesCompleteUploadExternalPostResponsedefault|FilesCompleteUploadExternalPostResponse200|null {
+        $filesPayload = [];
+
+        foreach ($files as $file) {
+            $filePath = $file['path'];
+            $fileName = basename($filePath);
+            $title = $file['title'] ?? $fileName;
+            $altText = $file['alt_text'] ?? null;
+            $snippetType = $file['snippet_type'] ?? null;
+
+            $fileStream = Stream::create(fopen($filePath, 'r'));
+            $fileSize = $fileStream->getSize();
+
+            // Step 1: Get upload URL from Slack
+            $queryParams = array_filter(
+                [
+                    'filename' => $fileName,
+                    'length' => $fileSize,
+                    'alt_txt' => $altText,
+                    'snippet_type' => $snippetType,
+                ]
+            );
+
+            $uploadInfo = $this->filesGetUploadUrlExternal($queryParams);
+            $uploadUrl = $uploadInfo?->getUploadUrl();
+            $fileId = $uploadInfo?->getFileId();
+
+            if (null === $uploadUrl || null === $fileId) {
+                throw new \RuntimeException("Slack did not return upload URL or file ID for {$fileName}");
+            }
+
+            // Step 2: Upload file data using cURL
+            $this->uploadToSlackUrl($uploadUrl, $filePath);
+
+            // Step 3: Build file metadata
+            $filesPayload[] = [
+                'id' => $fileId,
+                'title' => $title,
+            ];
+        }
+
+        // Step 4: Complete upload
+        return $this->filesCompleteUploadExternal(
+            array_filter(
+                [
+                    'files' => json_encode($filesPayload, \JSON_THROW_ON_ERROR),
+                    'channel_id' => $channelId,
+                    'initial_comment' => $initialComment,
+                ]
+            )
+        );
+    }
+
+    /**
+     * @throws ClientExceptionInterface
+     */
+    private function uploadToSlackUrl(string $uploadUrl, string $filePath): void
+    {
+        $uriFactory = Psr17FactoryDiscovery::findUriFactory();
+        $uri = $uriFactory->createUri($uploadUrl);
+
+        $streamFactory = Psr17FactoryDiscovery::findStreamFactory();
+        $stream = $streamFactory->createStreamFromFile($filePath);
+
+        $requestFactory = Psr17FactoryDiscovery::findRequestFactory();
+        $request = $requestFactory->createRequest('POST', $uri);
+        $request = $request->withBody($stream);
+
+        $client = Psr18ClientDiscovery::find();
+        $response = $client->sendRequest($request);
```

Comment:
We must use the Client provided by the developer, so I think this should be:

```suggestion
        $response = $this->httpClient->sendRequest($request);
```

The HttpClient is from here: https://github.com/jolicode/slack-php-api/blob/92f86407ac9a3cad46985e608ec9a9514b5b2388/generated/Runtime/Client/Client.php#L31

Reply:
That's exactly what I did, it doesn't work at least when I run my test :(

Reply:
@damienalexandre it doesn't work because generated client has base uri set as slack.com and what getUploadUrlExternal returns is a different url i.e., `https://files.slack.com/upload/v1/CwABAAAAYwoAAdZbBYEretOHCgACGDH-HcmojZAMAAMLAAEAAAALRTA3UEI5R1QzTU0LAAIAAAALVTA4S1YyNjkwOEsLAAMAAAALRjA4S1kyQVRFUTYACgAEAAAAAAAAAJoIAAcAAAAEAAsAAgAAABTBhf2hsqHZcysRfVuQ1zXp61dVZAA` and that's why I am getting 404s

Reply:
Ok I see, I just tested your branch, and found a way to keep the "files.slack.com" URI when using the internal HTTP Client. I just pushed a new commit in your PR to add this. Is it ok for you?

Reply:
Nice, looks good! 👍